pkg/controller/argocd/role.go

```diff
@@ -107,20 +107,30 @@ func (r *ReconcileArgoCD) reconcileRole(name string, policyRules []v1.PolicyRule
 }
 
 func (r *ReconcileArgoCD) reconcileClusterRole(name string, policyRules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD) (*v1.ClusterRole, error) {
-
 	clusterRole := newClusterRole(name, policyRules, cr)
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRole.Name}, clusterRole)
+	clusterRole.Rules = policyRules
+	applyReconcilerHook(cr, clusterRole, "")
+
+	existingClusterRole := &v1.ClusterRole{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRole.Name}, existingClusterRole)
 	if err != nil {
 		if !errors.IsNotFound(err) {
 			return nil, fmt.Errorf("failed to reconcile the cluster role for the service account associated with %s : %s", name, err)
 		}
-		applyReconcilerHook(cr, clusterRole, "")
-		return clusterRole, r.client.Create(context.TODO(), newClusterRole(name, policyRules, cr))
+		if len(clusterRole.Rules) == 0 {
+			// Do Nothing
+			return nil, nil
+		}
+		return clusterRole, r.client.Create(context.TODO(), clusterRole)
 	}
 
-	clusterRole.Rules = policyRules
-	applyReconcilerHook(cr, clusterRole, "")
-	return clusterRole, r.client.Update(context.TODO(), clusterRole)
+	if len(clusterRole.Rules) == 0 {
+		return nil, r.client.Delete(context.TODO(), existingClusterRole)
+	}
+
+	existingClusterRole.Rules = policyRules
+	applyReconcilerHook(cr, existingClusterRole, "")
```

Comment:
`applyReconcilerHook` is called the second time to policyRules.  Openshift policy rules will get appended again.

Reply:
`applyReconcilerHook` is being called on different objects. But I agree we don't need to call it again. I can directly substitute the rules!! Thanks, I'll update this!

Reply:
Different objects (clusterRole and existingClusterRole) but both objects embed the same policyRules if I read this right.  So, policyRules ends up with 2 x injected rules.

Reply:
Done !!! fixed it